test/expr/unary/keypath/keypath.swift

```diff
@@ -893,6 +893,34 @@ struct SR_12290 {
   }
 }
 
+// SR-12745
+struct SR12745 {
+  var x : Int = 0
+}
+
+struct SR12745_1 {
+  var x : Int = 0
+
+  subscript(_ i: String) -> Int { 0 }
+}
+
+func f(_ i: Int) {}
+
+func testKeyPathHole(_ s: SR12745) {
+  s[\.x] = 10 // expected-error {{cannot infer key path type from context; consider adding a root type}}
+  // expected-error@-1 {{value of type 'SR12745' has no matching subscript candidates; did you mean to use a key path subscript?}} {{5-5=keyPath: }}
+  _ = \.x // expected-error {{cannot infer key path type from context; consider adding a root type}}
+  s[\SR12745.x] // expected-error {{value of type 'SR12745' has no matching subscript candidates; did you mean to use a key path subscript?}} {{5-5=keyPath: }}
+
+  let _ : AnyKeyPath = \.x // expected-error {{cannot infer key path type from context; consider adding a root type}}
+
+  func f(_ i: Int) {}
+  f(\.x) // expected-error {{cannot infer key path type from context; consider adding a root type}}
```

Comment:
Not sure if we should diagnose anything else here since `repairFailures` is considering an `ApplyArgToParam` failure involving a hole as fixed ...
https://github.com/apple/swift/blob/7d2f9440866c5360cc4929f003850058ecfc98bc/lib/Sema/CSSimplify.cpp#L3619
cc @xedin